sys/riscv/riscv/locore.S

```diff
@@ -229,7 +231,13 @@ va:
 	sd	t0, RISCV_BOOTPARAMS_KERN_STACK(sp)
 
 	li	t0, (VM_EARLY_DTB_ADDRESS)
+	/* Add offset of DTB relative to start of superpage */
+	srli	t1, a1, (L2_SHIFT)
+	slli	t1, t1, (L2_SHIFT)
+	sub	t1, a1, t1
+	add	t0, t0, t1
```

Comment:
```suggestion
	/* Add offset of DTB within superpage */
	li	t1, (L2_OFFSET)
	and	t1, a1, t1
	add	t0, t0, t1
```

Reply:
Same number of instructions (I think?) but a bit clearer. Applied.

Reply:
Yeah, though if we had the B extension we could `li t1, ~(L2_OFFSET) ; and t1, a1, t1` and the `li` could use a sign-extending `lui` :). I haven't had a look at which form is more compressible and wouldn't like to say, but don't really care about instruction count in locore...